k2/csrc/fsa_util.cc

```diff
@@ -6,9 +6,91 @@
 
 #include "k2/csrc/fsa_util.h"
 
+#include <algorithm>
+#include <cctype>
+#include <cstdlib>
 #include <utility>
 #include <vector>
 
+#include "glog/logging.h"
+
+namespace {
+
+int32_t StringToInt(const std::string &s, bool *is_ok = nullptr) {
+  CHECK_EQ(s.empty(), false);
+
+  bool ok = false;
+  char *p = nullptr;
+  // std::strtol requires a `long` type
+  long n = std::strtol(s.c_str(), &p, 10);
+  if (*p == 0) ok = true;
```

Comment:
`if (*p == '\0')`?

Reply:
0 is identical to `\0` with less typing.

Reply:
check section `0 and nullptr/NULL` in [google style](https://google.github.io/styleguide/cppguide.html)

Reply:
thanks, will update it in the next commit.

Reply:
fixed in #23